### DISCUSSION TITLE
Show all unchanged lines

### DISCUSSION THREAD

**Discussion by Zebradil** [dandavison/delta#1646] [Q&A] [answered]
How to instruct delta to output all lines when diffing two files?
This only shows differences with some context around them:

```shell
delta 1.txt 2.txt
```

For example, `diff -y 1.txt 2.txt` does show all lines from the files.

**Comment by th1000s** [comment, marked as answer]
`delta 1.txt 2.txt` calls an external diff tool, but the passed arguments can't be changed. But you can call this tool yourself, and then pipe that to `delta`:

`git diff --no-index -U9999 1.txt 2.txt | delta`
or
 `diff -U9999 1.txt 2.txt | delta`

**Reply by Zebradil** [reply]
I didn't know that you can specify a context size with the diff's `-U` flag. Thanks!

**Reply by dandavison** [reply]
Thanks @th1000s. I've opened a PR https://github.com/dandavison/delta/pull/1697 adding an option `--diff-args=...` (short form `-@=...`) allowing extra arguments to `git diff` to be passed from the delta command line, as suggested by @calestyo in https://github.com/dandavison/delta/issues/1644#issuecomment-1974396731.
